main.cpp

```diff
@@ -61,124 +69,411 @@ New/This/Pointers/References conclusion
 
 
 
-#include <math.h> // I needed to do an ABS function...
+#include <math.h> // I needed to do an ABS function... and a twoExp
 #include <iostream> // to print a divide-by-zero console message
 
+//forward declarations of other types for overloaded member functions
+struct DoubleType;
+struct IntType;
+
 struct FloatType
 {
-    float add (float, float);
-    float subtract ( float, float);
-    float multiply ( float, float);
-    float divide (float, float);
+    float* pointerToFloatValue = nullptr;
+
+    // default constructor assigns heap primitive if no value sent in
+    FloatType() {
+        pointerToFloatValue = new float();
+     } 
+
+    FloatType(float floatIn )
+    {
+        pointerToFloatValue = new float( floatIn );
+    }
+    
+    ~FloatType()
+    {
+        delete pointerToFloatValue;
+        pointerToFloatValue = nullptr;
+    }
+   
+    FloatType& add ( float );
+    FloatType& subtract ( float );
+    FloatType& multiply ( float );
+    FloatType& divide ( float );
+    FloatType& add ( const DoubleType& );
+    FloatType& subtract ( const DoubleType& );
+    FloatType& multiply ( const DoubleType& );
+    FloatType& divide ( const DoubleType& );
+    FloatType& add ( const IntType& );
+    FloatType& subtract ( const IntType& );
+    FloatType& multiply ( const IntType& );
+    FloatType& divide ( const IntType& );
 };
 
-float FloatType::add (float x, float y)
+FloatType& FloatType::add ( const float x )
 {
-    return (x+y);
+    *pointerToFloatValue += x;
+    return *this;
 }
 
-float FloatType::subtract (float x, float y) 
+FloatType& FloatType::subtract ( const float y ) 
 { 
-    return (x-y); 
+    *pointerToFloatValue -= y;
+    return *this; 
 }
 
-float FloatType::multiply (float x, float y) 
+FloatType& FloatType::multiply ( const float y) 
 { 
-    return (x*y); 
+    *pointerToFloatValue *= y;
+    return *this; 
 }
 
-float FloatType::divide (float x, float y) 
+FloatType& FloatType::divide ( const float y ) 
 { 
-    if (abs(y) > 0) return (x/y);    
+    if (abs(y) > 0) *pointerToFloatValue /= y;    
     
     std::cout << "Divide-by-zero warning  " << std::endl;
-    return (x/y);      
+    *pointerToFloatValue /= 0;
+    return *this;      
```

Comment:
you need an `else` here.  you're always dividing by 0. 

- [ ] add `else` block, remove division by zero.  floating point division by zero is allowed.